File: tests/java/tests-java-junit5-api_request-playwright/src/test/java/api/MockScenarios.java
package api;

import io.qameta.allure.Step;

import java.util.Map;

/**
 * WireMock scenario switch for the mock stand (compose profile {@code mock}; the gateway
 * proxies {@code /__admin/} to WireMock). Lets UI tests inject API failures that a healthy
 * live backend can never produce.
 */
public final class MockScenarios {

    private MockScenarios() {
    }

    /** True when the stand under test exposes the WireMock admin API (mock stand only). */
    public static boolean available() {
        try {
            return PlaywrightHttp.request("GET", "/__admin/scenarios").status() == 200;
        } catch (RuntimeException gatewayUnreachable) {
            return false;
        }
    }

    @Step("Mock: switch scenario {scenario} to state {state}")
    public static void setState(String scenario, String state) {
        HttpResult response = PlaywrightHttp.request(
                "PUT",
                "/__admin/scenarios/" + scenario + "/state",
                Map.of("state", state),
                null,
                null);
        if (response.status() != 200) {
            throw new IllegalStateException(response.body());
        }
    }

    @Step("Mock: reset all scenarios")
    public static void resetAll() {
        HttpResult response = PlaywrightHttp.request("POST", "/__admin/scenarios/reset");
        if (response.status() != 200) {
            throw new IllegalStateException(response.body());
        }
    }
}

File: tests/kotlin/tests-kotlin-junit5-api_request-playwright/src/test/kotlin/api/MockScenarios.kt
package api

import com.microsoft.playwright.PlaywrightException
import io.qameta.allure.Step

/**
 * WireMock scenario switch for the mock stand (compose profile `mock`; the gateway
 * proxies `/__admin/` to WireMock). Lets UI tests inject API failures that a healthy
 * live backend can never produce.
 *
 * In-cell HTTP is Playwright APIRequest — not a second folder.
 */
object MockScenarios {

    /** True when the stand under test exposes the WireMock admin API (mock stand only). */
    @JvmStatic
    fun available(): Boolean {
        return try {
            PlaywrightHttp.request("GET", "/__admin/scenarios").status == 200
        } catch (_: PlaywrightException) {
            false
        } catch (_: RuntimeException) {
            false
        }
    }

    @JvmStatic
    @Step("Mock: switch scenario {scenario} to state {state}")
    fun setState(scenario: String, state: String) {
        val response = PlaywrightHttp.request(
            "PUT",
            "/__admin/scenarios/$scenario/state",
            json = mapOf("state" to state),
        )
        check(response.status == 200) { response.body }
    }

    @JvmStatic
    @Step("Mock: reset all scenarios")
    fun resetAll() {
        val response = PlaywrightHttp.request("POST", "/__admin/scenarios/reset")
        check(response.status == 200) { response.body }
    }
}
